llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -1274,6 +1274,8 @@ AArch64TargetLowering::AArch64TargetLowering(const TargetMachine &TM,
       setOperationAction(ISD::AVGCEILU, VT, Legal);
       setOperationAction(ISD::ABDS, VT, Legal);
       setOperationAction(ISD::ABDU, VT, Legal);
+      setOperationAction(ISD::TRUNCATE_SSAT, VT, Legal);
```

Comment:
Is the intent to say that a `MVT::v8i16 -> MVT::v8i8` truncating saturate is legal? If so does it need to mark the `MVT::v8i16` or the `MVT::v8i8`, and what happens if it is a `MVT::v8i32 -> MVT::v8i8` or `MVT::v4i32 -> MVT::v4i8` truncate?

If the isOperationLegalOrCustom looks at SrcVT then we likely only need to mark `MVT::v8i16, MVT::v4i32, MVT::v2i64` as legal, but there should hopefully be tests for the other case mentioned above. Note that I don't think that v2i64->v2i32 is handled at the moment with our tablegen patterns because we don't have support for 64bit min/max. So hopefully they might be able to start working with this patch.

Reply:
@davemgreen got it. I will change it to custom so that it is handled properly when legalizing the operand.